be/src/exec/parquet_reader.cpp

```diff
@@ -41,27 +41,53 @@ namespace starrocks {
 
 // Broker
 
-ParquetReaderWrap::ParquetReaderWrap(FileReader* file_reader, int32_t num_of_columns_from_file)
+ParquetReaderWrap::ParquetReaderWrap(FileReader* file_reader, int32_t num_of_columns_from_file, int64_t read_offset,
+                                     int64_t read_size)
         : ParquetReaderWrap(std::shared_ptr<arrow::io::RandomAccessFile>(new ParquetFile(file_reader)),
-                            num_of_columns_from_file) {}
+                            num_of_columns_from_file, read_offset, read_size) {}
+
+ParquetReaderWrap::~ParquetReaderWrap() {
+    close();
+}
 
 ParquetReaderWrap::ParquetReaderWrap(std::shared_ptr<arrow::io::RandomAccessFile>&& parquet_file,
-                                     int32_t num_of_columns_from_file)
+                                     int32_t num_of_columns_from_file, int64_t read_offset, int64_t read_size)
+
         : _num_of_columns_from_file(num_of_columns_from_file),
           _total_groups(0),
           _current_group(0),
           _rows_of_group(0),
           _current_line_of_group(0),
-          _current_line_of_batch(0) {
+          _current_line_of_batch(0),
+          _read_offset(read_offset),
+          _read_size(read_size) {
     _parquet = std::move(parquet_file);
     _properties = parquet::ReaderProperties();
     _properties.enable_buffered_stream();
     _properties.set_buffer_size(8 * 1024 * 1024);
 }
 
-ParquetReaderWrap::~ParquetReaderWrap() {
-    close();
+Status ParquetReaderWrap::next_selected_row_group() {
+    while (_current_group < _total_groups) {
+        int64_t row_group_start = _file_metadata->RowGroup(_current_group)->file_offset();
+
+        if (_read_size == 0) {
+            return Status::OK();
```

Comment:
should return EndOfFile?

Reply:
_read_size == 0 is to be compatible with the case where the scan range is the default value, that is, the read size is 0, but this does lead to misleading interfaces for ParquetReaderWrap，I will try to fix it.